app/src/ActionModal/components/ModalVoteSubstrate.vue

```diff
@@ -0,0 +1,428 @@
+    return {
+      lockedBalance: {
+        required,
+        decimal,
+        max: (x) => Number(x) <= this.stakingDenomBalance.available,
+        min: (x) => Number(x) >= SMALLEST,
+        maxDecimals: (x) => {
+          if (x) {
+            return x.toString().split(".").length > 1
+              ? x.toString().split(".")[1].length <= 6
+              : true
+          } else {
+            return false
+          }
+        },
+      },
+      vote: {
+        required,
+        isValid,
+      },
+    }
+  },
+  methods: {
+    open() {
+      this.$refs.actionModal.open()
+    },
+    validateForm() {
+      this.$v.$touch()
+
+      return !this.$v.$invalid
+    },
+    clear() {
+      this.$v.$reset()
+
+      this.vote = null
+    },
+    onSuccess(event) {
+      this.$emit(`success`, event)
+    },
+    lockingOptionController(lockingOption) {
+      this.selectedLockingOption = lockingOption
+      this.totalVotingController()
+      switch (lockingOption.display) {
+        case `0.1x`:
+          return (this.lockingPeriod = "0")
+        case `1x`:
+          return (this.lockingPeriod = "28")
+        case `2x`:
+          return (this.lockingPeriod = "56")
+        case `3x`:
+          return (this.lockingPeriod = "112")
+        case `4x`:
+          return (this.lockingPeriod = "224")
+        case `5x`:
+          return (this.lockingPeriod = "448")
+        case `6x`:
+          return (this.lockingPeriod = "896")
+        case `Set Max`:
+          return (this.lockingPeriod = "896")
+        default:
+          return (this.lockingPeriod = "0")
+      }
+    },
+    totalVotingController() {
+      this.totalVotingPower = (
+        this.lockedBalance * this.selectedLockingOption.multiplier
+      ).toFixed(6)
+    },
+  },
+  apollo: {
+    balances: {
+      query: gql`
+        query($networkId: String!, $address: String!) {
+          balancesV2(networkId: $networkId, address: $address) {
+            id
+            denom
+            available
+            total
+          }
+        }
+      `,
+      /* istanbul ignore next */
+      variables() {
+        return {
+          networkId: this.currentNetwork.id,
+          address: this.address,
+        }
+      },
+      /* istanbul ignore next */
+      skip() {
+        return !this.address
+      },
+      update(result) {
+        this.lockedBalance = result.balancesV2.find(
+          ({ denom }) => denom === this.currentNetwork.stakingDenom
+        ).available
+        this.totalVotingPower = this.lockedBalance
+        return result.balancesV2
+      },
+    },
+  },
+}
+</script>
+
+<style scoped>
+.buttons {
+  display: flex;
+  justify-content: right;
+}
+
+.card {
+  border: solid 1px var(--app-fg-hover);
+  padding: 1rem;
+  border-radius: 5px;
+  margin-right: 0.5rem;
+}
+
+.card-title {
+  display: block;
+  font-size: var(--sm);
+  color: var(--input-bc);
+}
+
+.line {
+  border: solid 1px var(--app-fg-hover);
+}
+
+.flex-row {
```

Comment:
@jbibla should we have such classes in the components or globally?

Reply:
we don't use a lot of global styles. this is fine especially because it's scoped.